### PR TITLE
Add snapshot endpoints

### PR DESCRIPTION
**Pull request by mariobodemann**
see https://www.contentful.com/developers/docs/references/content-management-api/#/reference/snapshots/entry-snapshots-collection

### CODE REVIEW

File: src/main/java/com/contentful/java/cma/ModuleEntries.java [hunk 1]
```diff
@@ -227,6 +228,46 @@ public CMAEntry update(CMAEntry entry) {
...
+    final String entryId = getResourceIdOrThrow(entry, "entry");
+    final String spaceId = getSpaceIdOrThrow(entry, "entry");
+
+    return service.fetchAllSnapshots(spaceId, entryId).toBlocking().first();
+  }
+
+  /**
+   * Fetch a specific snapshot of an entry.
+   *
+   * @param entry      the entry whose snapshot to be returned.
```
Comment (Roblinde) [inline src/main/java/com/contentful/java/cma/ModuleEntries.java]: Spacing bonanza?

File: src/main/java/com/contentful/java/cma/ModuleEntries.java [hunk 2]
```diff
@@ -414,5 +455,47 @@ public Async async() {
...
+        @Override CMAArray<CMASnapshot> method() {
+          return ModuleEntries.this.fetchAllSnapshots(entry);
+        }
+      }, callback);
+    }
+
+    /**
+     * Fetch a specific snapshot of an entry.
+     *
+     * @param entry      the entry whose snapshot to be returned.
```
Comment (Roblinde) [inline src/main/java/com/contentful/java/cma/ModuleEntries.java]: Spacing bonanza number 2?

File: src/main/java/com/contentful/java/cma/ModuleEntries.java [hunk 3]
```diff
@@ -227,6 +228,46 @@ public CMAEntry update(CMAEntry entry) {
   }
 
   /**
+   * Fetch all snapshots of an entry.
+   *
+   * @param entry the entry whose snapshots to be returned.
+   * @return an array of snapshots.
+   * @throws IllegalArgumentException if entry is null.
+   * @throws IllegalArgumentException if entry 's id is null.
+   * @throws IllegalArgumentException if entry 's space id is null.
```
Comment (dborsatto) [inline src/main/java/com/contentful/java/cma/ModuleEntries.java]: Total nitpick, but there's an unnecessary space between `entry` and `'s` ;)

Comment (Roblinde) [inline src/main/java/com/contentful/java/cma/ModuleEntries.java]: YES! Moar spaces!

File: src/main/java/com/contentful/java/cma/ModuleEntries.java (hunk 1 above)
Comment (mariobodemann) [inline src/main/java/com/contentful/java/cma/ModuleEntries.java]: That is to align the description of the `entry` param with the description of the `snapshotId`

File: src/main/java/com/contentful/java/cma/ModuleEntries.java (hunk 3 above)
Comment (mariobodemann) [inline src/main/java/com/contentful/java/cma/ModuleEntries.java]: Great spot! I 'll fix that in a second ... ;)

Comment (dborsatto) [inline src/main/java/com/contentful/java/cma/ModuleEntries.java]: It's also somewhere else below, the same typo

File: src/main/java/com/contentful/java/cma/ModuleEntries.java [hunk 4]
```diff
@@ -414,5 +455,47 @@ public Async async() {
...
+
+    /**
+     * Fetch all snapshots of an entry.
+     *
+     * @param entry the entry whose snapshots to be returned.
+     * @return an array of snapshots.
+     * @throws IllegalArgumentException if entry is null.
+     * @throws IllegalArgumentException if entry 's id is null.
+     * @throws IllegalArgumentException if entry 's space id is null.
+     */
```
Comment (dborsatto) [inline src/main/java/com/contentful/java/cma/ModuleEntries.java]: ☝️

**Pull request merged**